### PR TITLE
Update cosmos to federated auth

### PR DESCRIPTION
**Pull request by weshaggard**
Update cosmos test pipeline configurations to use federated auth.

### CODE REVIEW

File: sdk/cosmos/kafka.yml [hunk 1]
```diff
@@ -7,7 +7,6 @@ extends:
       - template: /eng/pipelines/templates/stages/archetype-sdk-tests-isolated.yml
         parameters:
           TestName: 'Kafka_TestContainer_Integration'
-          UseFederatedAuth: false
           EnvVars:
             ACCOUNT_TENANT_ID: '72f988bf-86f1-41af-91ab-2d7cd011db47'
             ACCOUNT_AAD_CLIENT_ID: $(spark-databricks-cosmos-spn-clientId)
```
Comment (weshaggard) [inline sdk/cosmos/kafka.yml:12]: @nehrao1 @kushagraThapar you should please work to remove this client id and client secret.

Comment (kushagraThapar) [reply sdk/cosmos/kafka.yml:12]: Ack.

**Comment by weshaggard** [comment]
[Cosmos test pipeline](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4275209&view=results)

**Review by joshfree — APPROVED** [review]

File: sdk/cosmos/kafka.yml (hunk 1 above)
Comment (tvaron3) [reply sdk/cosmos/kafka.yml:12]: We use this for our kafka connector live tests. I think this should be separated as our kafka connector does not support managed identity yet.

Comment (xinlian12) [reply sdk/cosmos/kafka.yml:12]: yea it is used in java - cosmos - kafka

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**